### ISSUE TITLE
Force Training

### ISSUE THREAD

**Issue by vsumaria** [ulissigroup/amptorch#81]
I'm trying to train an alloy system model. I'm more interested in getting the forces accurate. But while training using amptorch, I am facing the following issue - As energy MAE decreases, Force MAE starts increasing, even though the total loss is decreasing. I'm attaching the output convergence file. 

[out.log](https://github.com/ulissigroup/amptorch/files/5937981/out.log)

I tried to increase the force_coefficient, change the learning rate, also increases the number of BP descriptors - none of the solutions are helping. Any advise would be great. 

Thanks

**Comment by mshuaibii** [comment]
Hi @vsumaria,

Can you share your training `config`, it may help give better advise. This is a little tricky though, just to make sure...is the underlying ground truth you're trying to model  energy/force consistent, this could be part of the reason if not. 

Otherwise, modifying the scaling scheme used, maybe `"scaling": {"type": "normalize", "range": (-1, 1)}` can also help here. What sort of force coefficients have you explored, if you were only interested in forces I would try setting this to 100-1000 if you haven't done so already.

- Muhammed

**Comment by vsumaria** [comment]
Here's the config: 

Gs = {
    "default": {
        "G2": {
            "etas": [1.0, 0.5, 0.22, 0.125, 0.08, 0.056, 0.041, 0.031, 0.025, 0.0165],
            "rs_s": [0]
        },
        "G4": {"etas": [0.0165, 0.0473, 0.5], "zetas": [1.0, 4.0, 6.0], "gammas": [1.0, -1.0]},
        "cutoff": 6.0,
    },
}

config = {
    "model": {
        "get_forces": True,
	"num_layers": 2,
        "num_nodes": 5,
        "batchnorm": False,
       "activation": Sigmoid
    },
    "optim": {
        "force_coefficient": 0.1,
        "lr": 0.1,
        "batch_size": 64,
        "epochs": 10000,
        "loss": "mse",
	"metric": "mae",
        "gpus": 0,
        "scheduler": {"policy": "StepLR", "params": {"step_size": 100, "gamma": 0.5}}
    },
    "dataset": {
        "raw_data": images,
        "val_split": 0.1,
        "fp_params": Gs,
        "save_fps": True,
	# feature scaling to be used - normalize or standardize
	# normalize requires a range to be specified
        "scaling": {"type": "normalize", "range": (0, 1)},
    },
    "cmd": {
        "debug": False,
        "run_dir": "./",
        "seed": 1,
        "identifier": "test",
        "verbose": True,
        # Weights and Biases used for logging - an account(free) is required
        "logger": False,
    },
}

I only tried force coefficients at max of 0.3, but it makes sense to make that higher to give more importance to the force convergence. I'll also check the influence on the scaling.

**Comment by vsumaria** [comment]
@mshuaibii I think both these suggestions were great. The force errors are still not as low as I would want, but they are significantly lower that what I was getting initially. I think next step would to be optimize the symmetry functions. 

But for now, am closing the comment. Thanks again.
